### ISSUE TITLE
json can't decode UTF8 encoded string

### ISSUE THREAD

**Issue by asakura** [erlang/otp#8246]
**Describe the bug**
json's utf8 validator doesn't recognize ø as a valid character so `json:decode/1` fails with `invalid_byte` error.

**To Reproduce**
```
1> json:decode(<<"\"ø\"">>).
** exception error: {invalid_byte,248}
     in function  json:invalid_byte/2 (json.erl, line 534)
     in call from json:decode/1 (json.erl, line 624)
2> json:decode(<<"\"abc\"">>).
<<"abc">>
```

**Expected behavior**
Should return a decoded string.

**Affected versions**
OTP27

**Additional context**
Add any other context about the problem here. If you wish to attach Erlang code you can either write it directly in the post using [code tags](https://guides.github.com/features/mastering-markdown/), create a [gist](https://gist.github.com), or attach it as a zip file to this post.

**Comment by asakura** [comment]
cc @michalmuskala

**Comment by dgud** [comment]
The binary you created isn't not a utf8 binary, try:
json:decode(<<"\"ø\""/utf8>>).

**Comment by dgud** [comment]
Or nowadays:   json:decode(~' "ø" ').

**Comment by essen** [comment]
Isn't it just you trying to decode latin1 and not Unicode? You need to add the `/utf8` to your example.

```
1> <<A>> = <<"ø">>, A.
248
2> <<B,C>> = <<"ø"/utf8>>, {B, C}.
{195,184}
```

**Comment by michalmuskala** [comment]
Yes, the regular binary syntax (without `/utf8`) has the somewhat unfortunate behaviour of limiting every character to 255 (with overflow) so the binary `<<"\"ø\"">>` ends up being `<<34,248,34>>`, rather than the expected `<<34,195,184,34,0>>`.

The new `~""` binary syntax fixes this "gotcha" in a great way.
